File: Inheritence.py
# Base class
class Person:
    def __init__(self, name):
        self.name = name

    def talk(self):
        print(f"{self.name} is talking.")

# Derived class from Person
class Chef(Person):
    def cook(self):
        print(f"{self.name} is cooking delicious food.")

# Derived class from Chef (multi-level inheritance)
class PastryChef(Chef):
    def bake_cake(self):
        print(f"{self.name} is baking a beautiful cake.")

# Example usage
renu = PastryChef("Renuka")

renu.talk()         # Inherited from Person
renu.cook()         # Inherited from Chef
renu.bake_cake()    # Defined in PastryChef

File: Inheritence.cpp
#include <iostream>
using namespace std;

// Base class
class Person {
public:
    string name;

    Person(string name) {
        this->name = name;
    }

    void talk() {
        cout << name << " is talking." << endl;
    }
};

// Derived class from Person
class Chef : public Person {
public:
    Chef(string name) : Person(name) {}

    void cook() {
        cout << name << " is cooking delicious food." << endl;
    }
};

// Derived class from Chef (Multi-level Inheritance)
class PastryChef : public Chef {
public:
    PastryChef(string name) : Chef(name) {}

    void bakeCake() {
        cout << name << " is baking a beautiful cake." << endl;
    }
};

// Main function
int main() {
    PastryChef renu("Renuka");

    renu.talk();       // From Person
    renu.cook();       // From Chef
    renu.bakeCake();   // From PastryChef

    return 0;
}
